index.html

```diff
@@ -3812,7 +3821,7 @@ <h2>The <dfn>ThingDiscoveryProcess</dfn> interface</h2>
     <pre class="idl">
       [SecureContext, Exposed=(Window,Worker)]
       interface ThingDiscoveryProcess {
-        constructor(optional ThingFilter filter = {});
+        constructor(optional ThingFilter filter = {}, optional URL url);
```

Comment:
This change would need to be reflected in the TypeScript definition also, right?

https://github.com/w3c/wot-scripting-api/blob/23dc59e121abf582a285709fd9aa917344c7fcbf/typescript/scripting-api/index.d.ts#L63-L68

Reply:
> Maybe add a separate method with a mandatory URL argument. 

Actually, per the TAG API design principles spec, we should avoid constructor overloading, so if we need UA magic, we should use factory methods to create a `ThingDiscoveryProcess`.

Reply:
I moved the commit with the changes now to a different branch, so we could discuss the API changes here separately. I will open another PR for that.

Reply:
Not sure if the changes now present in this PR actually reflect our discussion we had in #535 (yet). If not, maybe we can even close this PR and start from scratch.